crowbar_framework/lib/crowbar/upgrade.rb

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Upgrade
+    attr_accessor :data
+
+    def initialize(backup)
+      @backup = backup
+      @data = @backup.data
+      @version = @backup.version
+    end
+
+    def upgrade
+      knife_files
+    end
+
+    def supported?
+      upgrades = [
+        ["1.9", "3.0"]
+      ]
+      upgrades.include?([@version, ENV["CROWBAR_VERSION"]])
+    end
+
+    protected
+
+    def knife_files
+      FileUtils.mv(@data.join("knife", "data_bags"), @data.join("knife", "databags"))
+      FileUtils.rm_rf(@data.join("knife", "databags", "barclamps"))
+
+      crowbar_databags_path = @data.join("knife", "databags", "crowbar")
+      Dir.entries(crowbar_databags_path).each do |file|
+        if file == "bc-template-nova_dashboard.json"
+          File.rename(crowbar_databags_path.join(file), crowbar_databags_path.join(file.sub!("nova_dashboard", "horizon")))
```

Comment:
Metrics/LineLength: Line is too long. [123/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)